tensorflow/core/kernels/conv_ops_gpu_3.cu.cc

```diff
@@ -413,6 +429,275 @@ struct PadInput<GPUDevice, T, int, NDIMS> {
+                         TileLongSide == 128  && (TileShortSide OP 15) || \
+                         TileLongSide == 256  && (TileShortSide OP 10) || \
+                         TileLongSide == 512  && (TileShortSide OP 4)  || \
+                         TileLongSide == 1024 && (TileShortSide OP 2)) || \
+      size_of_t == 1 && (TileLongSide == 32   && (TileShortSide OP 15) || \
+                         TileLongSide == 64   && (TileShortSide OP 15) || \
+                         TileLongSide == 128  && (TileShortSide OP 15) || \
+                         TileLongSide == 256  && (TileShortSide OP 10) || \
+                         TileLongSide == 512  && (TileShortSide OP 4)  || \
+                         TileLongSide == 1024 && (TileShortSide OP 2))
+
+constexpr bool TileSizePossibilityFrontierCheck(int TileLongSide,
+                                                int TileShortSide,
+                                                int size_of_t, int mode) {
+  // When mode is 0, this function checks whether the tile size combination lies
+  // on the tile size possibility frontier. When mode is 1, it checks whether
+  // the combination lies outside the frontier.
+  return mode == 0 ? TPF_CHECKER(==) : TPF_CHECKER(>);
+}
+
+#undef TPF_CHECKER
+
+// Recursive template function to search for the minimum tile size configuration
+// satisfying the requested tile side lengths.
+template <typename T, int TileLongSide, int TileShortSide, typename dummy = void>
+struct BatchNarrowMatrixTransposeDispatcher {
+  static void DoIt(const GPUDevice& d, int tile_size_i,
+                                           int tile_size_j,
+                                           int total_tiles_count,
+                                           const T* input,
+                                           const Dimension<3>& input_dims,
+                                           T* output) {
+
+    static_assert((TileLongSide & (TileLongSide - 1)) == 0,
+      "The length of the longer side of the tile is always a power of 2.");
+    bool request_satisfied = max(tile_size_i, tile_size_j) <= TileLongSide &&
+                             min(tile_size_i, tile_size_j) <= TileShortSide;
+
+    if (request_satisfied) {
+      const int NumThread = TileLongSide;
+      if (tile_size_i <= TileLongSide && tile_size_j <= TileShortSide) {
+        SwapDimension1And2InTensor3UsingTiles<
+            T, NumThread, TileLongSide,
+            TileShortSide><<<total_tiles_count, NumThread, 0, d.stream()>>>(
+            input, input_dims, output);
+      }
+      else if (tile_size_j <= TileLongSide && tile_size_i <= TileShortSide) {
+        SwapDimension1And2InTensor3UsingTiles<
+            T, NumThread, TileShortSide,
+            TileLongSide><<<total_tiles_count, NumThread, 0, d.stream()>>>(
+            input, input_dims, output);
+      }
+      return;
+    }
+
+    // Kernel is not launched, meaning the launch configuration is not
```

Comment:
s/is/was/?

Reply:
Maybe we can be more precise in the part of the comment after the comma.  Right now it just repeats the fact that if we got here, `!request_satisfied`.  Fine, but what does that really mean?